File: LeetCode/Math/JS/Leetcode762-二进制表示中的质数置位.js
var countPrimeSetBits = function(left, right) {
    const primes = [],st = new Array(34).fill(false);
    let get_primes = function(n) {
        st[1] = true;
        for(let i = 2;i <= n;i++) {
            if(!st[i]) primes.push(i);
            for(let j = 0;primes[j] <= Math.floor(n / i);j++) {
                st[primes[j] * i] = true;
                if(i % primes[j] == 0) break;
            }
        }
    }
    let check = function(n) {
        let cnt = 0;
        while(n) {
            n -= (n & -n);
            cnt++;
        }
        return !st[cnt];
    }
    get_primes(33);
    let ans = 0;
    for(let i = left;i <= right;i++) {
        if(check(i)) ans++;
    }
    return ans;
};  

File: LeetCode/Math/Golang/Leetcode762-二进制表示中的质数置位.go
package sbgo

func countPrimeSetBits(left int, right int) int {
	st, primes := make([]bool, 33), []int{}
	get_primes := func(n int) {
		st[1] = true
		for i := 2; i <= n; i++ {
			if !st[i] {
				primes = append(primes, i)
			}
			for j := 0; primes[j] <= n/i; j++ {
				st[primes[j]*i] = true
				if i%primes[j] == 0 {
					break
				}
			}
		}
	}
	get_primes(32)
	ans := 0
	check := func(n int) bool {
		cnt := 0
		for n != 0 {
			cnt++
			n -= (n & -n)
		}
		return !st[cnt]
	}
	for i := left; i <= right; i++ {
		if check(i) {
			ans++
		}
	}
	return ans
}
